### DISCUSSION TITLE
classes/class_staticbody2d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#161] [User-contributed notes]
# classes/class_staticbody2d

Inherits: PhysicsBody2D< CollisionObject2D< Node2D< CanvasItem< Node< Object Inherited By: AnimatableBody2D A 2D physics body that can't be moved by external forces. When moved manually, it doesn't...

https://docs.godotengine.org/en/stable/classes/class_staticbody2d.html

**Comment by lemonhall** [comment]
remember to use this,for collision

```gdscript
func _physics_process(delta):
	velocity = Vector2.ZERO # The player's movement vector.
	if Input.is_action_pressed("ui_up",true):
		animated_sprite_2d.play("up")
		velocity.y = -speed
	if Input.is_action_pressed("ui_down",true):
		animated_sprite_2d.play("face_down")
		velocity.y = speed
	if Input.is_action_pressed("ui_left",true):
		animated_sprite_2d.play("left")
		velocity.x = -speed
	if Input.is_action_pressed("ui_right",true):
		animated_sprite_2d.play("right")
		velocity.x = speed
	if velocity.length() > 0:
		move_and_slide()
	pass
```

move_and_slide()!!!!!

this takes me 2 hours to work

**Reply by Wiechciu** [reply]
But move_and_slide() method exists on the CharacterBody2D, not StaticBody2D.
https://docs.godotengine.org/en/stable/classes/class_characterbody2d.html#class-characterbody2d-method-move-and-slide

**Comment by NancokPS2** [comment]
Note: at least since 4.2 (possibly since the introduction of AnimatableBody) Areas do NOT detect moving StaticBodies.  
This is not stated anywhere.

**Reply by miketvo** [reply]
Thank you, this is exactly what I was looking for! Saves me a lot of headache as I was scratching my head wondering why my moving StaticBodies are not being detected.
